enumerables.rb

```diff
@@ -0,0 +1,160 @@
+module Enumerable
+
+  # my_each
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_each
+    return to_enum(:my_each) unless block_given?
+      i = 0
+      while i < self.size do
+        yield(self.to_a[i])
+        i += 1
+      end
+  end
+  
+  # my_each_with_index
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+    i = 0
+    while i < self.length do
+      yield(i,self[i])
+      i += 1
+    end
+  end
```

Comment:
Let's fix the issue with `my_each_with_index`, see the difference with the ruby's method below, using the example from the ruby doc( link below):

```ruby
hash = Hash.new
%w(cat dog wombat).each_with_index { |item, index|
  hash[item] = index
}
# => ["cat", "dog", "wombat"]
p hash
# => {"cat"=>0, "dog"=>1, "wombat"=>2}
hash = Hash.new
%w(cat dog wombat).my_each_with_index { |item, index|
  hash[item] = index
}
# => nil
p hash
# => {0=>"cat", 1=>"dog", 2=>"wombat"}
```

Two things to consider:
1. `yield` argument should be interchanged, the second argument is the index while the first is the collection member. See https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-each_with_index , this way the final result will not be interchanged as seen in the final result (`hash`)  in the example above.
2. Let's return(implicit return) `self`, this way the result of running `my_each_with_index` would be `["cat", "dog", "wombat"]` instead on `nil` in this example.